libirods_rule_engine_plugin-elasticsearch.cpp

```diff
@@ -1,913 +1,805 @@
+				auto it = _args.begin();
+				const std::string object_path{boost::any_cast<std::string>(*it)};
+				++it;
+				const std::string attribute{boost::any_cast<std::string>(*it)};
+				++it;
+				const std::string value{boost::any_cast<std::string>(*it)};
+				++it;
+				const std::string unit{boost::any_cast<std::string>(*it)};
+				++it;
+				const std::string index_name{boost::any_cast<std::string>(*it)};
+				++it;
+
+				std::string obj_meta_str = "{}";
+
+				if (it != _args.end()) {
+					obj_meta_str = boost::any_cast<std::string>(*it++);
+				}
+
+				json obj_meta = nlohmann::json::parse(obj_meta_str);
+
+				// purge with AVU by name?
+				if (_rn == metadata_purge_policy && attribute.empty()) {
+					// delete the indexed entry
+					invoke_purge_event_metadata(rei, object_path, attribute, value, unit, index_name);
+				}
+				else {
+					// update the indexed entry
+					invoke_indexing_event_metadata(rei, object_path, attribute, value, unit, index_name, obj_meta);
+				}
+			}
+			else if (_rn == "irods_policy_recursive_rm_object_by_path") {
+				auto it = _args.begin();
+				const std::string the_path{boost::any_cast<std::string>(*it)};
+				std::advance(it, 2);
+				const json recurse_info = json::parse(boost::any_cast<std::string&>(*it));
+
+				const auto escaped_path = [p = the_path]() mutable {
+					boost::replace_all(p, "\\", "\\\\");
+					boost::replace_all(p, "?", "\\?");
+					boost::replace_all(p, "*", "\\*");
+					return p;
+				}();
+
+				std::string JtopLevel = json{{"query", {{"match", {{"absolutePath", escaped_path}}}}}}.dump();
+				std::string JsubObject;
+
+				try {
+					if (recurse_info["is_collection"].get<bool>()) {
+						JsubObject =
+							json{{"query", {{"wildcard", {{"absolutePath", {{"value", escaped_path + "/*"}}}}}}}}
+								.dump();
+					}
+				}
+				catch (const std::domain_error& e) {
+					// TODO What is this about? Why std::domain_error?
+					return ERROR(
+						-1,
+						fmt::format("_delete_by_query - stopped short of performRequest - domain_error: {}", e.what()));
+				}
+
+				try {
+					for (const std::string& index_name : recurse_info["indices"]) {
+						for (const std::string& json_out : {JtopLevel, JsubObject}) {
+							if (json_out.empty()) {
+								continue;
+							}
+
+							const auto response = send_http_request(
+								http::verb::post, fmt::format("{}/_delete_by_query", index_name), json_out);
+
+							if (!response.has_value()) {
+								rodsLog(LOG_ERROR,
+								        fmt::format("{}: No response from elaticsearch host.", __func__).c_str());
+								continue;
+							}
+
+							if (response->result_int() != 200) {
+								rodsLog(LOG_WARNING,
+								        fmt::format("_delete_by_query - response code not 200"
+								                    "\n\t- for path [{}]"
+								                    "\n\t- escaped as [{}]"
+								                    "\n\t- json request body is [{}]",
```

Comment:
consider no tabs in the logging... perhaps just structured logging is good enough.

Reply:
I'm planning on handling all log messages in a separate PR.